### DISCUSSION TITLE
Why is `IValueDidChange` using unknown-property values?

### DISCUSSION THREAD

**Discussion by misantronic** [mobxjs/mobx#3517] [General] [answered]
The current type-interface for `IValueDidChange` looks like this:

```ts
type IValueDidChange<T = any> = {
    type: "update";
    observableKind: "value";
    object: IObservableValue<T>;
    debugObjectName: string;
    newValue: unknown;
    oldValue: unknown;
}
```

In an older version of mobx, this was typed so that `newValue` and `oldValue` would use `T | undefined` as their default type which in my opinion makes total sense. Why is it now `unknown`?

**Comment by emereum** [comment]
This change also impacts us in a large codebase. It seems like this was done out of concern that a value could escape the type system, but I don't see how that is possible given `IObservableValue<T>`'s very narrow API surface:

```
export interface IObservableValue<T> {
    get(): T;
    set(value: T): void;
    intercept_(handler: IInterceptor<IValueWillChange<T>>): Lambda;
    observe_(listener: (change: IValueDidChange<T>) => void, fireImmediately?: boolean): Lambda;
}
```

It would be helpful if the maintainers can advise what risk was mitigated by this change. This information will help steer our implementation of additional runtime type checking to resolve the new type errors.

**Comment by mweststrate** [comment, marked as answer]
The `unknown` looks really strange indeed, and I can't find any history back that this was intentionally changed, so feel to change it back. I'm wondering if this is strictness implemented wrong, and that the actual thing that should have happened is changing the generic default from `T = any` to `T = unknown`. Feel free to submit a PR.

**Reply by emereum** [reply]
PR submitted #3609
